app/assets/javascripts/admin/addon/components/schema-theme-setting/types/models.gjs

```diff
@@ -0,0 +1,43 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { action } from "@ember/object";
+import I18n from "discourse-i18n";
+
+export default class SchemaThemeSettingTypeModels extends Component {
+  @tracked touched = false;
+  @tracked value = this.args.value;
+
+  required = this.args.spec.required;
+  min = this.args.spec.validations?.min;
+  max = this.args.spec.validations?.max;
+  type;
+
+  @action
+  onInput(newValue) {
+    this.touched = true;
+    this.value = newValue;
+    this.args.onChange(this.onChange(newValue));
+  }
+
+  onChange(newValue) {
+    return newValue;
+  }
+
+  get validationErrorMessage() {
+    if (!this.touched) {
+      return;
+    }
+
+    if (
+      (this.min && this.value.length < this.min) ||
```

Comment:
Just wondering, do you use `min/max` HTML attrs on the actual input fields?

Also, you are not checking for `this.max` here either.

Reply:
>  do you use min/max HTML attrs on the actual input fields?

Yup I do for input fields but for Discourse models based field components, we rely on the various selectkit components. 

> Also, you are not checking for this.max here either.

This is intentional because we pass `this.max` as the `maximum` option to the select-kit components which makes it impossible for the user to select more than the max and hence why there is no need to display a validation message.